### PR TITLE
Fix `napari.utils.misc.is_iterable` function

### PR DESCRIPTION
**Pull request by Czaki**
# Description

During development of #6780 I have found that `napari.utils.misc.is_iterable` function wrongly return that object is iterable, when it does not have `__iter__` method. This PR fixes it and is extracted from #6870 to reduce diff. 

This PR adds basic tests, fix handling affine in split_channels (as affine could be `Affine` or `np.ndarray`).

### CODE REVIEW

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -922,7 +922,9 @@ def add_image(
             'translate',
             'rotate',
             'shear',
-            'affine',
+            'affine',  # affine may be Affine or np.ndarray object that is not
+            # iterable, but it is not now a problem as we use it only to warning
+            # if a provided object is sequence and channel_axis is not provided
```
Comment (brisvag) [inline napari/components/viewer_model.py]: I would move this comment to the if/else block in the other file

File: napari/layers/utils/stack_utils.py [hunk 2]
```diff
@@ -122,6 +121,8 @@ def split_channels(
                     allow_none=True,
                 )
             )
+        elif key == 'affine' and isinstance(val, np.ndarray):
+            kwargs[key] = itertools.repeat(val, n_channels)
         else:
```
Comment (brisvag) [inline napari/layers/utils/stack_utils.py]: comment here that affine `np.ndarray` is a special case and why

File: napari/utils/misc.py [hunk 3]
```diff
@@ -149,16 +149,16 @@ def is_iterable(
...
-    if arg is None and not allow_none:
-        return False
+    if arg is None:
+        return allow_none
     if isinstance(arg, (str, Enum)):
         return False
     if np.isscalar(arg):
         return False
     if color and isinstance(arg, (list, np.ndarray)):
         return np.array(arg).ndim != 1 or len(arg) not in [3, 4]
```
Comment (brisvag) [inline napari/utils/misc.py:160]: actually, should we special case affine/np.ndarray in here, like we did with colors?

File: napari/utils/_tests/test_misc.py [hunk 4]
```diff
@@ -251,3 +252,18 @@ def test_ensure_list_of_layer_data_tuple(input_data, expected):
...
+        (1, False),
+        (1.0, False),
+        ([1], True),
+        ('aaa', False),
+        (object(), False),
+        (None, False),
+    ],
+)
+def test_is_iterable(data, expected):
+    assert is_iterable(data) == expected
```
Comment (brisvag) [inline napari/utils/_tests/test_misc.py]: can/should we add tests for stuff like `np.ndarray` and so on?

File: napari/utils/misc.py (hunk 3 above)
Comment (Czaki) [reply napari/utils/misc.py:160]: We do not use color anywhere

**Review by brisvag — APPROVED** [review]

**Pull request merged**